src/configs/FirebaseConfig.js

```diff
@@ -59,6 +60,23 @@ const FirebaseConfig = (props) => {
     }
   };
 
+  const getUserProfileV2 = async (url) => {
+    if (url && typeof url === 'string') {
+      const userId = await getUserId();
+      const parts = url.split('/profile/');
+      const urlSplit = parts?.[1];
+      const paramSplit = urlSplit?.split('?userId=');
+      const username = paramSplit?.[0];
+      const targetUserId = paramSplit?.[1];
+      const data = {
+        other_id: targetUserId,
+        username,
+        user_id: userId
+      };
+      handleMovePage(USER, data);
+    }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `getUserProfileV2` seems to be parsing the URL manually. This can lead to errors if the URL structure changes in the future. It would be better to use a URL parsing library or built-in functionality to extract parameters from the URL. Here's an example using the built-in `URL` and `URLSearchParams` classes:

```diff
-      const parts = url.split('/profile/');
-      const urlSplit = parts?.[1];
-      const paramSplit = urlSplit?.split('?userId=');
-      const username = paramSplit?.[0];
-      const targetUserId = paramSplit?.[1];
+      const urlObj = new URL(url);
+      const username = urlObj.pathname.replace('/profile/', '');
+      const params = new URLSearchParams(urlObj.search);
+      const targetUserId = params.get('userId');
```
This approach is more robust and less error-prone than manual string splitting.


<!-- This is an auto-generated comment by OSS CodeRabbit -->